contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/HttpClientHandler.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.http.core;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.opentelemetry.contrib.http.core.HttpTraceConstants.INSTRUMENTATION_LIB_ID;
+
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.distributedcontext.DistributedContext;
+import io.opentelemetry.distributedcontext.DistributedContextManager;
+import io.opentelemetry.distributedcontext.Entry;
+import io.opentelemetry.distributedcontext.EntryMetadata.EntryTtl;
+import io.opentelemetry.metrics.Meter;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.Tracer;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * This helper class provides routine methods to instrument HTTP servers.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @param <C> the type of the carrier.
+ */
+public class HttpClientHandler<Q, P, C> extends AbstractHttpHandler<Q, P> {
+
+  private static final Logger LOGGER = Logger.getLogger(HttpClientHandler.class.getName());
+
+  private final Tracer tracer;
+  private final DistributedContextManager contextManager;
+  private final HttpTextFormat.Setter<C> setter;
+  private final HttpTextFormat<SpanContext> spanContextHttpTextFormat;
+  private final HttpTextFormat<DistributedContext> distributedContextHttpTextFormat;
+
+  /**
+   * Constructs a handler object.
+   *
+   * @param extractor used to extract information from request/response
+   * @param setter the setter used to extract information from the carrier
```

Comment:
If a `setter` is used to extract information, then it's definitely named incorrectly.

Reply:
Fixed